src/App.js

```diff
@@ -25,44 +25,97 @@ const generateSquares = () => {
   return squares;
 }
 
-const App = () => {
 
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
+const App = () => {
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState('')
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
-
+  const onClickCallback = (id) => {
+    let row = 0;
+    while (id > 2) {
+      id -= 3
+      row ++
+    }
+  }
```

Comment:
you are not calling this function anywhere. For the onClickCallBack prop, you are passing in the updateSquares function. So this can be removed.